File: app/src/main/java/de/tum/in/tumcampus/models/Cafeteria.java
package de.tum.in.tumcampus.models;

/**
 * Cafeteria Object
 */
public class Cafeteria implements Comparable<Cafeteria> {
	/**
	 * Address, e.g. Boltzmannstr. 3
	 */
	public String address;

	/**
	 * Cafeteria ID, e.g. 412
	 */
	public int id;

	/**
	 * Name, e.g. MensaX
	 */
	public String name;

    public double latitude;
    public double longitude;

    // Used for ordering cafeterias
    public float distance;

	/**
	 * new Cafeteria
	 * 
	 * <pre>
	 * @param id Cafeteria ID, e.g. 412
	 * @param name Name, e.g. MensaX
	 * @param address Address, e.g. Boltzmannstr. 3
     * @param latitude Coordinates of the cafeteria
     * @param longitude Coordinates of the cafeteria
	 * </pre>
	 */
	public Cafeteria(int id, String name, String address, double latitude, double longitude) {
		this.id = id;
		this.name = name;
		this.address = address;
        this.latitude = latitude;
        this.longitude = longitude;
	}

	@Override
	public String toString() {
		return name;
	}

    @Override
    public int compareTo(Cafeteria cafeteria) {
        return distance<cafeteria.distance?-1:1;
    }
}

File: app/src/main/java/de/tum/in/tumcampusapp/component/ui/cafeteria/model/Cafeteria.kt
package de.tum.`in`.tumcampusapp.component.ui.cafeteria.model

import androidx.room.Entity
import androidx.room.PrimaryKey
import androidx.room.RoomWarnings
import com.google.gson.annotations.SerializedName

/**
 * new Cafeteria
 *
 * @param id Cafeteria ID, e.g. mensa-garching
 * @param slug cafeteria string identification slug e.g.: "mensa-garching"
 * @param name Name, e.g. MensaX
 * @param address Address, e.g. Boltzmannstr. 3
 * @param latitude Coordinates of the cafeteria
 * @param longitude Coordinates of the cafeteria
 */
@Entity
@SuppressWarnings(RoomWarnings.DEFAULT_CONSTRUCTOR)
data class Cafeteria(
        @PrimaryKey
        var id: Int = 0,
        var slug: String = "",
        var name: String = "",
        var address: String = "",
        var latitude: Double = -1.0,
        var longitude: Double = -1.0
) : Comparable<Cafeteria> {

    // Used for ordering cafeterias
    var distance: Float = 0f

    override fun toString(): String = name

    override fun compareTo(other: Cafeteria): Int = distance.compareTo(other.distance)
}